domain/src/test/java/org/oppia/domain/profile/ProfileManagementControllerTest.kt

```diff
@@ -783,13 +753,259 @@ class ProfileManagementControllerTest {
+      assertThat(deviceSettings.automaticallyUpdateTopics).isFalse()
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testDeviceSettings_addAdminProfile_updateDeviceWifiSettings_getDeviceSettings_isSuccessful() =
+    runBlockingTest(coroutineContext) {
+      profileManagementController.addProfile(
+        name = "James",
+        pin = "12345",
+        avatarImagePath = null,
+        allowDownloadAccess = true,
+        colorRgb = -10710042,
+        isAdmin = true,
+        storyTextSize = StoryTextSize.SMALL_TEXT_SIZE,
+        appLanguage = AppLanguage.ENGLISH_APP_LANGUAGE,
+        audioLanguage = AudioLanguage.ENGLISH_AUDIO_LANGUAGE
+      ).observeForever(mockUpdateResultObserver)
+      advanceUntilIdle()
+
+      val adminProfileId = ProfileId.newBuilder().setInternalId(0).build()
+      profileManagementController.updateWifiPermissionDeviceSettings(
+        adminProfileId, /* downloadAndUpdateOnWifiOnly = */
+        true
+      ).observeForever(mockUpdateResultObserver)
+      advanceUntilIdle()
+
+      verifyUpdateSucceeded()
+
+      profileManagementController.getDeviceDettings().observeForever(mockDeviceSettingsObserver)
+      advanceUntilIdle()
+
+      verifyGetDeviceSettingsSucceeded()
+      val deviceSettings = deviceSettingsResultCaptor.value.getOrThrow()
+      assertThat(deviceSettings.allowDownloadAndUpdateOnlyOnWifi).isTrue()
+      assertThat(deviceSettings.automaticallyUpdateTopics).isFalse()
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testDeviceSettings_addAdminProfile_updateTopicsAutomaticallyDeviceSettings_getDeviceSettings_isSuccessful() =
+    runBlockingTest(coroutineContext) {
+      profileManagementController.addProfile(
+        name = "James",
+        pin = "12345",
+        avatarImagePath = null,
+        allowDownloadAccess = true,
+        colorRgb = -10710042,
+        isAdmin = true,
+        storyTextSize = StoryTextSize.SMALL_TEXT_SIZE,
+        appLanguage = AppLanguage.ENGLISH_APP_LANGUAGE,
+        audioLanguage = AudioLanguage.ENGLISH_AUDIO_LANGUAGE
+      ).observeForever(mockUpdateResultObserver)
+      advanceUntilIdle()
+
+      val adminProfileId = ProfileId.newBuilder().setInternalId(0).build()
+      profileManagementController.updateTopicAutomaticallyPermissionDeviceSettings(
+        adminProfileId, /* automaticallyUpdateTopics = */
+        true
+      )
+        .observeForever(mockUpdateResultObserver)
+      advanceUntilIdle()
+
+      verifyUpdateSucceeded()
+
+      profileManagementController.getDeviceDettings().observeForever(mockDeviceSettingsObserver)
+      advanceUntilIdle()
+
+      verify(mockDeviceSettingsObserver, atLeastOnce()).onChanged(deviceSettingsResultCaptor.capture())
+      assertThat(deviceSettingsResultCaptor.value.isSuccess()).isTrue()
+
+      val deviceSettings = deviceSettingsResultCaptor.value.getOrThrow()
+      assertThat(deviceSettings.allowDownloadAndUpdateOnlyOnWifi).isFalse()
```

Comment:
We need to test auto update of topics here.

Reply:
Actually I am checking both values in each test case making sure that one value update does not affect the other.

Reply:
Makes sense.